File: petclinic-java-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/java/dev/pollito/petclinic_java_gradle_thymeleaf_tailwind_webpack_htmx_ts/visit/VisitService.java
package dev.pollito.petclinic_java_gradle_thymeleaf_tailwind_webpack_htmx_ts.visit;

import java.util.List;

public interface VisitService {

    List<Visit> findAll();

    Visit get(Integer id);

    Integer create(Visit visit);

    void update(Integer id, Visit visit);

    void delete(Integer id);
}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/visit/VisitService.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.visit

interface VisitService {

    fun findAll(): List<VisitDTO>

    fun `get`(id: Int): VisitDTO

    fun create(visitDTO: VisitDTO): Int

    fun update(id: Int, visitDTO: VisitDTO)

    fun delete(id: Int)
}
